src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/SystemModulesPlugin.java

```diff
@@ -696,6 +696,51 @@ private void genIncubatorModules(ClassWriter cw) {
          * Generate bytecode for moduleDescriptors method
          */
         private void genModuleDescriptorsMethod(ClassWriter cw) {
+            int helperMethodCount = 0;
+            final String helperMethodNamePrefix = "moduleDescriptorsSub";
+
+            // Implementation note:
+            // Here the method is "manually split" based on the heuristics that 99 ModuleDescriptors are smaller than 64kb
+            // The other implementation possibility is to use msplit (https://github.com/cretz/msplit). However, this seemed too much effort.
+            for (int index = 0; index < moduleInfos.size(); index++) {
```

Comment:
This comment will need to be cleaned up a bit to remove the reference to the "msplit" project and changed to be more consistent with the existing comments.

Reply:
I found the existing comments in the code pretty short. I am belonging to the "architectural decisions group" strengthening the need of listing "options" one choose from. See https://adr.github.io/madr/ for more information.

I will delete the comment.